rtos/rtx/TARGET_CORTEX_M/cmsis_os.h

```diff
@@ -88,8 +88,13 @@
 #if defined(TARGET_XDOT_L151CC)
 #define DEFAULT_STACK_SIZE         (WORDS_STACK_SIZE/2)
 #else
+#if defined (MBED_CFG_DEBUG_OPTIONS_COVERAGE)
+/* GCOV works with stack size approx. 32K. */
+#define DEFAULT_STACK_SIZE         (WORDS_STACK_SIZE*4*8)
```

Comment:
The heavy stack usage is only on the thread which writes out the coverage data. Also, if you turn on stack profiling you should be able to see exactly how much stack is used without the need for trial and error.